Tests/Realm.Tests/Sync/SessionTests.cs

```diff
@@ -1016,13 +1016,73 @@ public void Session_Free_Instance_Even_With_PropertyChanged_Subscribers()
                 var session = realm.SyncSession;
                 weakSessionRef = new WeakReference(session);
                 Assert.That(weakSessionRef.IsAlive, Is.True);
-                session.PropertyChanged += (sender, e) => { };
+                session.PropertyChanged += HandlePropertyChanged;
             });
 
+            static void HandlePropertyChanged(object? sender, PropertyChangedEventArgs e)
+            {
+            }
+
+            GC.Collect();
+            Assert.That(weakSessionRef.IsAlive, Is.True);
+        }
+
+        [Test]
+        public void Session_Should_Free_Instance_With_No_PropertyChanged_Subscribers()
+        {
+            WeakReference weakSessionRef = null!;
+
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var config = await GetIntegrationConfigAsync();
+                using var realm = GetRealm(config);
+                var session = realm.SyncSession;
+                weakSessionRef = new WeakReference(session);
+                Assert.That(weakSessionRef.IsAlive, Is.True);
+                session.PropertyChanged += HandlePropertyChanged;
+                session.PropertyChanged -= HandlePropertyChanged;
+            });
+
+            static void HandlePropertyChanged(object? sender, PropertyChangedEventArgs e)
+            {
+            }
+
             GC.Collect();
             Assert.That(weakSessionRef.IsAlive, Is.False);
         }
 
+        [Test]
+        public void AnotherTest()
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                WeakReference weakSessionRef = null!;
+
+                var config = await GetIntegrationConfigAsync();
+                using var realm = GetRealm(config);
+
+                weakSessionRef = new Func<WeakReference>(() => {
+                    return new WeakReference(realm.SyncSession);
+                })();
+
+                Assert.That(weakSessionRef.IsAlive, Is.True);
+
+                realm.SyncSession.PropertyChanged += HandlePropertyChanged;
+
+                GC.Collect();
+                Assert.That(weakSessionRef.IsAlive, Is.True);
+
+                realm.SyncSession.PropertyChanged -= HandlePropertyChanged;
+
+                GC.Collect();
+                Assert.That(weakSessionRef.IsAlive, Is.False);  //TODO This fails
```

Comment:
We do have a `WaitUntilReferencesAreCollected` method in TestHelpers that runs GC.Collect with retries - not sure why this test is failing but may be worth trying to use that instead.

Reply:
That worked nicely 😄